src/interpreter/inner.rs

```diff
@@ -274,18 +308,46 @@ pub fn from_txdata<'txin>(
     // ** bare script **
     } else {
         if wit_stack.is_empty() {
-            let miniscript = Miniscript::<bitcoin::PublicKey, _>::parse_insane(spk)?;
+            // Bare script parsed in BareCtx
+            let miniscript = Miniscript::<bitcoin::PublicKey, BareCtx>::parse_insane(spk)?;
+            let miniscript = pre_taproot_to_no_checks(&miniscript);
             Ok((
                 Inner::Script(miniscript, ScriptType::Bare),
                 ssig_stack,
-                spk.clone(),
+                Some(spk.clone()),
             ))
         } else {
             Err(Error::NonEmptyWitness)
         }
     }
 }
 
+// Convert a miniscript from a specified context into an insane miniscript
+// We need to remember how the hash160 was translated because while doing a checksig
+// we need to know whether to parse the public key provided in witness as x-only or full
+pub(super) fn pre_taproot_to_no_checks<Ctx: ScriptContext>(
+    ms: &Miniscript<bitcoin::PublicKey, Ctx>,
+) -> Miniscript<BitcoinKey, NoChecksEcdsa> {
+    // specify the () error type as this cannot error
+    ms.real_translate_pk::<_, _, _, (), _>(&mut |pk| Ok(BitcoinKey::Fullkey(*pk)), &mut |pkh| {
+        Ok(TaggedHash160::FullKey(*pkh))
+    })
+    .expect("Translation should succeed")
+}
+
+// Convert a miniscript from a specified context into an insane miniscript
+#[allow(dead_code)]
+pub(super) fn taproot_to_no_checks<Ctx: ScriptContext>(
```

Comment:
why is this conversion insane? why do we do this?

Reply:
Will add a comment elaborating this.